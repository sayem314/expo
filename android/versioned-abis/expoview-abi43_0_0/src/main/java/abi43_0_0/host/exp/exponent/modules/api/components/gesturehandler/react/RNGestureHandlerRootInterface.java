package abi43_0_0.host.exp.exponent.modules.api.components.gesturehandler.react;

import androidx.annotation.Nullable;

public interface RNGestureHandlerRootInterface {
  @Nullable RNGestureHandlerRootHelper getRootHelper();
}
