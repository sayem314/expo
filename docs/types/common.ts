export type PageMetadata = {
  title?: string;
  sourceCodeUrl?: string;
  packageName?: string;
  maxHeadingDepth?: number;
  /* If the page should not show up in the Algolia Docsearch results */
  hideFromSearch?: boolean;
  hideTOC?: boolean;
};

/**
 * A single header from the `remark-export-headings` plugin.
 */
export type RemarkHeading = {
  id?: string;
  depth: number;
  title: string;
  type: string;
};

/**
 * Utility type. Extracts `T` type from `T[]` array.
 */
export type ElementType<T extends any[]> = T extends (infer U)[] ? U : never;

export type Url = {
  pathname: string;
};

export type NavigationRoute = {
  as?: string;
  hidden: boolean;
  href: string;
  name: string;
  sidebarTitle?: string;
  weight?: number;
  children?: NavigationRoute[];
  posts?: NavigationRoute[];
};
