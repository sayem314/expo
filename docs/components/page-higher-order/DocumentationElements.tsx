import GithubSlugger from 'github-slugger';
import { useRouter } from 'next/router';
import React, { PropsWithChildren } from 'react';

import { HeadingManager } from '~/common/headingManager';
import DocumentationPage from '~/components/DocumentationPage';
import { HeadingsContext } from '~/components/page-higher-order/withHeadingManager';
import { PageMetadata, RemarkHeading } from '~/types/common';

type DocumentationElementsProps = PropsWithChildren<{
  meta: PageMetadata;
  headings: RemarkHeading[];
}>;

export default function DocumentationElements(props: DocumentationElementsProps) {
  const router = useRouter();
  const manager = new HeadingManager(new GithubSlugger(), {
    ...props.meta,
    headings: props.headings,
  });

  return (
    <HeadingsContext.Provider value={manager}>
      <DocumentationPage
        title={props.meta.title || ''}
        url={router}
        asPath={router.asPath}
        packageName={props.meta.packageName}
        sourceCodeUrl={props.meta.sourceCodeUrl}
        tocVisible={!props.meta.hideTOC}
        hideFromSearch={props.meta.hideFromSearch}>
        {props.children}
      </DocumentationPage>
    </HeadingsContext.Provider>
  );
}
