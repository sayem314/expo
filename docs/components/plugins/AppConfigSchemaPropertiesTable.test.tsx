import { render } from '@testing-library/react';
import GithubSlugger from 'github-slugger';
import * as React from 'react';

import AppConfigSchemaPropertiesTable, {
  formatSchema,
  createDescription,
  _getType,
  Property,
} from './AppConfigSchemaPropertiesTable';

import { HeadingManager } from '~/common/headingManager';
import { HeadingsContext } from '~/components/page-higher-order/withHeadingManager';

const testSchema: Record<string, Property> = {
  name: {
    description: 'Name of your app.',
    type: 'string',
    meta: {
      bareWorkflow: "Edit the 'Display Name' field in Xcode",
    },
  },
  androidNavigationBar: {
    description: 'Configuration for the bottom navigation bar on Android.',
    type: 'object',
    properties: {
      visible: {
        description: 'Determines how and when the navigation bar is shown.',
        type: 'string',
        properties: {
          always: {
            description: 'Test sub-sub-property',
            type: 'boolean',
          },
        },
        meta: {
          expoKit: 'Set this property using Xcode.',
        },
        enum: ['leanback', 'immersive', 'sticky-immersive'],
      },
      backgroundColor: {
        description: 'Specifies the background color of the navigation bar. ',
        type: 'string',
        pattern: '^#|(&#x23;)\\d{6}$',
        meta: {
          regexHuman: "6 character long hex color string, eg: `'#000000'`",
        },
      },
    },
    meta: {
      expoKit: 'Set this property using AppConstants.java.',
      bareWorkflow: 'Set this property using just Xcode',
    },
  },
  intentFilters: {
    description: 'Configuration for setting an array of custom intent filters in Android manifest.',
    example: [
      {
        autoVerify: true,
        data: {
          host: '*.example.com',
        },
      },
    ],
    exampleString: '\n [{ \n "autoVerify": true, \n "data": {"host": "*.example.com" \n } \n }]',
    type: 'array',
    uniqueItems: true,
    items: {
      type: 'object',
      properties: {
        autoVerify: {
          description:
            'You may also use an intent filter to set your app as the default handler for links',
          type: 'boolean',
        },
        data: {
          type: ['array', 'object'],
          items: {
            type: 'object',
            properties: {
              host: { description: 'the host, e.g. `example.com`', type: 'string' },
            },
            additionalProperties: false,
          },
          properties: {
            host: { type: 'string' },
          },
          additionalProperties: false,
        },
      },
      additionalProperties: false,
      required: ['action'],
    },
    meta: {
      bareWorkflow: 'This is set in AndroidManifest.xml directly.',
    },
  },
};

describe('AppConfigSchemaPropertiesTable', () => {
  test('correctly matches snapshot', () => {
    const { container } = render(
      <HeadingsContext.Provider value={new HeadingManager(new GithubSlugger(), { headings: [] })}>
        <AppConfigSchemaPropertiesTable schema={testSchema} />
      </HeadingsContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});

describe('formatSchema', () => {
  const formattedSchema = formatSchema(Object.entries(testSchema));
  test('name is property nestingLevel 0', () => {
    expect(formattedSchema[0].nestingLevel).toBe(0);
  });
  test('androidNavigationBar is property nestingLevel 0', () => {
    expect(formattedSchema[1].nestingLevel).toBe(0);
  });
  test('visible is subproperty nestingLevel 1', () => {
    expect(formattedSchema[2].nestingLevel).toBe(1);
  });
  test('always is subproperty nestingLevel 2', () => {
    expect(formattedSchema[3].nestingLevel).toBe(2);
  });
  test('backgroundColor is subproperty nestingLevel 1', () => {
    expect(formattedSchema[4].nestingLevel).toBe(1);
  });
  test('intentFilters is property nestingLevel 0', () => {
    expect(formattedSchema[5].nestingLevel).toBe(0);
  });
  test('autoVerify is subproperty nestingLevel 1', () => {
    expect(formattedSchema[6].nestingLevel).toBe(1);
  });
  test('data is subproperty nestingLevel 1', () => {
    expect(formattedSchema[7].nestingLevel).toBe(1);
  });
  test('host is subproperty nestingLevel 2', () => {
    expect(formattedSchema[8].nestingLevel).toBe(2);
  });
});

describe('createDescription', () => {
  test('bareWorkflow, exampleString are both added correctly to intentFilters', () => {
    const intentFiltersObject = Object.entries(testSchema)[2];
    const intentFiltersObjectValue = intentFiltersObject[1] as any;
    const result = createDescription(intentFiltersObject);

    expect(result).toBe(
      `**(${_getType(intentFiltersObjectValue)})** - ${
        intentFiltersObjectValue.description
      }<bareworkflowDetails>${
        intentFiltersObjectValue.meta!.bareWorkflow
      }</bareworkflowDetails>\n\n>${intentFiltersObjectValue.exampleString}`
    );
  });

  test('regexHuman is added correctly to backgroundColor', () => {
    //Note: to access this subproperty is tedious without a call to formatSchema
    const backgroundColorObject = Object.entries(Object.values(testSchema)[1].properties!)[1];
    const backgroundColorObjectValue = backgroundColorObject[1];
    const result = createDescription(backgroundColorObject);

    expect(result).toBe(
      `**(${_getType(backgroundColorObjectValue)})** - ${
        backgroundColorObjectValue.description
      }\n\n${backgroundColorObjectValue.meta!.regexHuman}`
    );
  });

  test('expoKit is added correctly to visible', () => {
    //Note: to access this subproperty is tedious without a call to formatSchema
    const visibleObject = Object.entries(Object.values(testSchema)[1].properties!)[0];
    const visibleObjectValue = visibleObject[1];
    const result = createDescription(visibleObject);

    expect(result).toBe(
      `**(${_getType(visibleObjectValue)})** - ${visibleObjectValue.description}<expokitDetails>${
        visibleObjectValue.meta!.expoKit
      }</expokitDetails>`
    );
  });

  test('bareWorkflow is added correctly to name', () => {
    const nameObject = Object.entries(testSchema)[0];
    const nameObjectValue = nameObject[1];
    const result = createDescription(nameObject);

    expect(result).toBe(
      `**(${_getType(nameObjectValue)})** - ${nameObjectValue.description}<bareworkflowDetails>${
        nameObjectValue.meta!.bareWorkflow
      }</bareworkflowDetails>`
    );
  });

  test('expoKit, bareWorkflow both added correctly to androidNavigationBar', () => {
    const androidNavigationBarObject = Object.entries(testSchema)[1];
    const androidNavigationBarObjectValue = androidNavigationBarObject[1] as any;
    const result = createDescription(androidNavigationBarObject);

    expect(result).toBe(
      `**(${_getType(androidNavigationBarObjectValue)})** - ${
        androidNavigationBarObjectValue.description
      }<expokitDetails>${
        androidNavigationBarObjectValue.meta!.expoKit
      }</expokitDetails><bareworkflowDetails>${
        androidNavigationBarObjectValue.meta!.bareWorkflow
      }</bareworkflowDetails>`
    );
  });
});
