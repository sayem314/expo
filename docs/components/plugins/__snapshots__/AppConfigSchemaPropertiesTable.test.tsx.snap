// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppConfigSchemaPropertiesTable correctly matches snapshot 1`] = `
<div>
  <table
    class="css-qctutf-STYLES_TABLE"
  >
    <thead
      class="css-1plmxco-STYLES_HEAD"
    >
      <tr>
        <td>
          Property
        </td>
        <td>
          Description
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            data-testid="<propertyAnchor level={0}><inlineCode>name</inlineCode></propertyAnchor>"
            style="margin-left: 0px; display: block; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="name"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#name"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      name
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (string)
            </strong>
             - Name of your app.
            <details
              class="css-lkrdcy-STYLES_DETAILS"
            >
              <summary
                class="css-642pov-STYLES_SUMMARY"
              >
                Bare Workflow
              </summary>
              <div
                class="css-1b7jivh-STYLES_CONTENTS"
              >
                Edit the 'Display Name' field in Xcode
              </div>
            </details>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<propertyAnchor level={0}><inlineCode>androidNavigationBar</inlineCode></propertyAnchor>"
            style="margin-left: 0px; display: block; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="androidnavigationbar"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#androidnavigationbar"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      androidNavigationBar
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (object)
            </strong>
             - Configuration for the bottom navigation bar on Android.
            <details
              class="css-lkrdcy-STYLES_DETAILS"
            >
              <summary
                class="css-642pov-STYLES_SUMMARY"
              >
                ExpoKit
              </summary>
              <div
                class="css-1b7jivh-STYLES_CONTENTS"
              >
                Set this property using AppConstants.java.
              </div>
            </details>
            <details
              class="css-lkrdcy-STYLES_DETAILS"
            >
              <summary
                class="css-642pov-STYLES_SUMMARY"
              >
                Bare Workflow
              </summary>
              <div
                class="css-1b7jivh-STYLES_CONTENTS"
              >
                Set this property using just Xcode
              </div>
            </details>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<subpropertyAnchor level={1}><inlineCode>visible</inlineCode></subpropertyAnchor>"
            style="margin-left: 32px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="visible"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#visible"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      visible
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (enum)
            </strong>
             - Determines how and when the navigation bar is shown.
            <details
              class="css-lkrdcy-STYLES_DETAILS"
            >
              <summary
                class="css-642pov-STYLES_SUMMARY"
              >
                ExpoKit
              </summary>
              <div
                class="css-1b7jivh-STYLES_CONTENTS"
              >
                Set this property using Xcode.
              </div>
            </details>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<subpropertyAnchor level={2}><inlineCode>always</inlineCode></subpropertyAnchor>"
            style="margin-left: 64px; display: list-item; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="always"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#always"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      always
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (boolean)
            </strong>
             - Test sub-sub-property
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<subpropertyAnchor level={1}><inlineCode>backgroundColor</inlineCode></subpropertyAnchor>"
            style="margin-left: 32px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="backgroundcolor"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#backgroundcolor"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      backgroundColor
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (string)
            </strong>
             - Specifies the background color of the navigation bar. 
          </div>
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            6 character long hex color string, eg: 
            <code
              class="inline css-nuv12a-STYLES_INLINE_CODE"
            >
              '#000000'
            </code>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<propertyAnchor level={0}><inlineCode>intentFilters</inlineCode></propertyAnchor>"
            style="margin-left: 0px; display: block; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="intentfilters"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#intentfilters"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      intentFilters
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (array)
            </strong>
             - Configuration for setting an array of custom intent filters in Android manifest.
            <details
              class="css-lkrdcy-STYLES_DETAILS"
            >
              <summary
                class="css-642pov-STYLES_SUMMARY"
              >
                Bare Workflow
              </summary>
              <div
                class="css-1b7jivh-STYLES_CONTENTS"
              >
                This is set in AndroidManifest.xml directly.
              </div>
            </details>
          </div>
          <blockquote
            class="css-e76wiv-paragraph-STYLES_BLOCKQUOTE"
            data-text="true"
          >
            <div>
              <div
                style="margin-top: 2px;"
              >
                <svg
                  aria-label="check"
                  fill="none"
                  height="16"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 0C5.373 0 0 5.373 0 12s5.373 12 12 12c6.628 0 12-5.373 12-12S18.628 0 12 0zm-.015 5.419c.269-.292.596-.44.973-.44.306 0 .575.11.776.317.203.207.306.472.306.786 0 .394-.137.745-.409 1.042-.279.306-.605.462-.968.462-.3 0-.566-.113-.768-.326-.201-.208-.304-.48-.304-.809 0-.401.133-.749.394-1.032zm1.999 11.305c-.995.936-1.705 1.52-2.17 1.784-.496.284-.905.423-1.25.423-.342 0-.621-.114-.83-.338-.202-.218-.305-.51-.305-.87 0-.92.521-3.058 1.595-6.534a1.47 1.47 0 00.042-.166c-.04.017-.087.041-.14.075-.088.055-.356.265-1.14 1.03a.334.334 0 01-.438.025l-.463-.358a.334.334 0 01-.043-.49c.722-.794 1.395-1.371 2.002-1.714.636-.358 1.173-.533 1.643-.533.303 0 .548.076.729.225.196.164.3.391.3.658 0 .175-.084.63-.697 2.686-.875 2.937-1.084 4.003-1.099 4.372.166-.098.558-.376 1.395-1.162a.334.334 0 01.462.003l.411.4a.334.334 0 01-.004.484z"
                    fill="var(--expo-theme-icon-default)"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
                data-text="true"
              >
                 
                [{
"autoVerify": true,
"data": {"host": "*.example.com"
}
}]
              </div>
            </div>
          </blockquote>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<subpropertyAnchor level={1}><inlineCode>autoVerify</inlineCode></subpropertyAnchor>"
            style="margin-left: 32px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="autoverify"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#autoverify"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      autoVerify
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (boolean)
            </strong>
             - You may also use an intent filter to set your app as the default handler for links
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<subpropertyAnchor level={1}><inlineCode>data</inlineCode></subpropertyAnchor>"
            style="margin-left: 32px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="data"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#data"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      data
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (array || object)
            </strong>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="<subpropertyAnchor level={2}><inlineCode>host</inlineCode></subpropertyAnchor>"
            style="margin-left: 64px; display: list-item; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
              data-text="true"
            >
              <div
                class="css-1rh3o5q-STYLES_PERMALINK"
              >
                <span
                  class="css-zonpcr-STYLES_PERMALINK_TARGET"
                  id="host"
                />
                <a
                  class="css-15lh5hg-STYLES_PERMALINK_LINK"
                  href="#host"
                >
                  <span
                    class="css-1icvi79-STYLED_PERMALINK_CONTENT"
                  >
                    <code
                      class="inline css-nuv12a-STYLES_INLINE_CODE"
                    >
                      host
                    </code>
                  </span>
                  <span
                    class="css-10vdcw8-STYLES_PERMALINK_ICON"
                  >
                    <svg
                      aria-label="permalink"
                      class="anchor-icon"
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                        stroke="#9B9B9B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-1pd7e0y-STYLES_DESCRIPTION_CELL"
        >
          <div
            class=" css-xq8ktq-paragraph-STYLES_PARAGRAPH_DIV"
            data-text="true"
          >
            <strong
              class="css-kkqcgt-paragraph-STYLES_BOLD_PARAGRAPH"
            >
              (string)
            </strong>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
